x-pack/filebeat/input/httpjson/httpjson_test.go

```diff
@@ -72,6 +69,43 @@ func runTest(t *testing.T, isTLS bool, m map[string]interface{}, run func(input
 			w.Write(b)
 		}
 	}))
+}
+
+func createCustomServer(newServer func(handler http.Handler) *httptest.Server) *httptest.Server {
+	return newServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
The `isRetry` state doesn't need to be a global. It could be 
```suggestion
	return newServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
	    var isRetry bool
```

Reply:
Putting it here actually will not work. It needs to go before the return statement.

Reply:
Yep, 😆 I thought that's where I did add it. woops